client/my-sites/site-settings/theme-setup/index.jsx

```diff
@@ -14,41 +14,71 @@ import QueryActiveTheme from 'components/data/query-active-theme';
 import QueryTheme from 'components/data/query-theme';
 import ThemeSetupCard from './theme-setup-card';
 import ThemeSetupPlaceholder from './theme-setup-placeholder';
-import { getSelectedSite } from 'state/ui/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
 import { getActiveTheme, getTheme } from 'state/themes/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
 import { toggleDialog } from 'state/ui/theme-setup/actions';
 
-let ThemeSetup = ( { site, themeId, theme, translate, activeSiteDomain, toggleDialog } ) => {
-	const onBack = () => {
-		page( '/settings/general/' + activeSiteDomain );
+class ThemeSetup extends Component {
+	componentDidUpdate( prevProps ) {
+		const {
+			siteIsJetpack,
+			siteId,
+		} = this.props;
+
+		if ( siteId !== prevProps.siteId && siteIsJetpack ) {
+			this.redirectToGeneral();
+		}
+	}
+
+	redirectToGeneral = () => {
+		const { siteSlug } = this.props;
+		page.redirect( '/settings/general/' + siteSlug );
 	};
 
-	return (
-		<div className="main theme-setup" role="main">
-			{ site && <QueryActiveTheme siteId={ site.ID } /> }
-			{ themeId && <QueryTheme siteId="wpcom" themeId={ themeId } /> }
-			<HeaderCake onClick={ onBack }><h1>{ translate( 'Theme Setup' ) }</h1></HeaderCake>
-			{ site && theme
-				? <ThemeSetupCard
-					onClick={ toggleDialog }
-					theme={ theme } />
-				: <ThemeSetupPlaceholder /> }
-		</div>
-	);
-};
+	render() {
+		const {
+			siteId,
+			theme,
+			themeId,
+			translate,
+		} = this.props;
 
-ThemeSetup = localize( ThemeSetup );
+		return (
+			<div className="main theme-setup" role="main">
```

Comment:
Indeed, good catch! Done in https://github.com/Automattic/wp-calypso/pull/15922/commits/ccf74e04b9a79a78bbe38e27d2a8003615c49251